### PR TITLE
test: migrate from tap to node:test and c8

### PR DESCRIPTION
**Pull request by inyourtime**
#### Checklist
This PR migrates from tap to node:test and c8

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/connection.test.js [hunk 1]
```diff
@@ -48,70 +48,64 @@ test('utils should work', (t) => {
...
     fastify.ready((err) => {
-      t.error(err)
+      t.assert.ifError(err)
 
       const sorter = 'date'
       const sql = 'SELECT * FROM posts ORDER BY ' + fastify.mysql.escapeId('posts.' + sorter)
-      t.ok(sql, 'SELECT * FROM posts ORDER BY `posts`.`date`')
-      t.end()
+      t.assert.strictEqual(sql, 'SELECT * FROM posts ORDER BY `posts`.`date`')
+      done()
```
Comment (inyourtime) [inline test/connection.test.js:103]: should be strictEqual

File: test/connection.test.js [hunk 2]
```diff
@@ -127,120 +121,116 @@ test('promise connection', (t) => {
...
```
Comment (inyourtime) [inline test/connection.test.js:174]: should be strictEqual

File: test/connection.test.js [hunk 3]
```diff
@@ -127,120 +121,116 @@ test('promise connection', (t) => {
...
```
Comment (inyourtime) [inline test/connection.test.js:185]: should be strictEqual

File: test/pool.test.js [hunk 4]
```diff
@@ -117,72 +121,75 @@ test('fastify.mysql.test namespace should exist', t => {
...
     })
 
-    test('mysql.escapeId', (t) => {
+    test('mysql.escapeId', (t, done) => {
       const sorter = 'date'
       const sql = 'SELECT * FROM posts ORDER BY ' + fastify.mysql.escapeId('posts.' + sorter)
-      t.ok(sql, 'SELECT * FROM posts ORDER BY `posts`.`date`')
-      t.end()
+      t.assert.strictEqual(sql, 'SELECT * FROM posts ORDER BY `posts`.`date`')
+      done()
```
Comment (inyourtime) [inline test/pool.test.js:172]: should be strictEqual

**Review by gurgunday — APPROVED** [review]

**Review by dancastillo — APPROVED** [review]
LGTM

**Pull request merged**